dspace-server-webapp/src/test/java/org/dspace/app/rest/DiscoveryScopeBasedRestControllerIT.java

```diff
@@ -0,0 +1,625 @@
+                   );
+
+        getClient().perform(get("/api/discover/facets/subcommunity11field")
+                                    .param("scope", String.valueOf(subcommunity11.getID())))
+                   .andExpect(status().isOk())
+                   .andExpect(jsonPath("$.type", is("discover")))
+                   .andExpect(jsonPath("$._embedded.values",
+                                       containsInAnyOrder(
+                                               FacetValueMatcher.matchEntry("subcommunity11field",
+                                                                            "subcommunity11field-item111", 1),
+                                               FacetValueMatcher.matchEntry("subcommunity11field",
+                                                                            "subcommunity11field-item112", 1),
+                                               FacetValueMatcher.matchEntry("subcommunity11field",
+                                                                            "subcommunity11field-mappedItem111222", 1)
+                                       )
+                   ));
+    }
+
+    @Test
+    public void ScopeBasedIndexingAndSearchTestCollection111() throws Exception {
+
+        getClient().perform(get("/api/discover/facets").param("scope", String.valueOf(collection111.getID())))
+
+                   .andExpect(status().isOk())
+                   .andExpect(jsonPath("$.type", is("discover")))
+                   .andExpect(jsonPath("$._links.self.href", containsString("api/discover/facets")))
+                   .andExpect(jsonPath("$._embedded.facets", containsInAnyOrder(
+                           FacetEntryMatcher.authorFacet(false),
+                           FacetEntryMatcher.matchFacet(false, "collection111field", "text")))
+                   );
+
+        getClient().perform(get("/api/discover/facets/collection111field")
+                                    .param("scope", String.valueOf(collection111.getID())))
+                   .andExpect(status().isOk())
+                   .andExpect(jsonPath("$.type", is("discover")))
+                   .andExpect(jsonPath("$._embedded.values",
+                                       containsInAnyOrder(
+                                               FacetValueMatcher.matchEntry("collection111field",
+                                                                            "collection111field-item111", 1),
+                                               FacetValueMatcher.matchEntry("collection111field",
+                                                                            "collection111field-mappedItem111222", 1)
+                                       )
+                   ));
+    }
+
+    @Test
+    public void ScopeBasedIndexingAndSearchTestCollection112() throws Exception {
+
+        getClient().perform(get("/api/discover/facets").param("scope", String.valueOf(collection112.getID())))
```

Comment:
It'd be very useful to add inline comments to these tests...to make it easier to understand the scenario each one is testing for.  E.g. This test seems to be testing for proper inheritance, as it inherits from `SubCommunity11`... while the SubCommunity11 test above is testing that a unique configuration in test-discovery.xml is loaded.

Having brief comments above each test which tell a developer what we are testing for would make this MUCH easier to read.  Otherwise, I have to have a few windows open and compare the data structure to what is in test-discovery.xml and also what is in this test. In other words, this is a bit of a maintenance nightmare as the tests are not self-describing... so we need to describe them with inline comments.